### PR TITLE
Refactor search index chunk updating

### PR DESCRIPTION
**Pull request by timheap**
Some chunks of functionality have been refactored out of the main flow of logic. Now it is easy to see what is going on and which bits of functionality relate to other bits. The behaviour is the same.

### CODE REVIEW

File: wagtail/wagtailsearch/management/commands/update_index.py [hunk 1]
```diff
@@ -116,3 +86,44 @@ def handle(self, **options):
...
+        of a ``for`` loop::
+
+            for item in self.progress(big_long_list):
+                self.do_expensive_computation(item)
+
+        A ``.`` character is printed for every value in the iterable,
+        a space every ``gap`` items, and a new line every ``new_line`` gaps.
+        Each line of output is prefixed with ``indent``
+        """
+        self.stdout.write(INDENT, ending='')
```
Comment (kaedroho) [inline wagtail/wagtailsearch/management/commands/update_index.py]: ``INDENT`` should be ``indent``

File: wagtail/wagtailsearch/management/commands/update_index.py [hunk 2]
```diff
@@ -116,3 +86,44 @@ def handle(self, **options):
...
+            if i % (gap * new_line) == 0:
+                self.stdout.write('')
+                self.stdout.write(INDENT, ending='')
+            elif i % gap == 0:
+                self.stdout.write(' ', ending='')
+            self.stdout.flush()
+        self.stdout.write('')
+
+    # Atomic so the count of models doesnt change as it is iterated
+    @transaction.atomic
```
Comment (kaedroho) [inline wagtail/wagtailsearch/management/commands/update_index.py]: Nice one :+1:

File: wagtail/wagtailsearch/management/commands/update_index.py (hunk 1 above)
Comment (timheap) [inline wagtail/wagtailsearch/management/commands/update_index.py]: Good catch, missed that in my rearranging. Fixed.

**Comment by kaedroho** [comment]
Thanks!

Just merged and made a few tweaks: https://github.com/kaedroho/wagtail/commit/e8e625c9d7a55bfcf93056642af39f10e24503b8

I've ditched the indenting. I think it might be nicer to use styling (like Djangos migrate command does) instead.

**Comment by timheap** [comment]
Looks good to me. Colours certainly do make things much clearer than an indent. Nice work

**Pull request merged**